examples/hazelcast/src/test/java/org/testcontainers/examples/HazelcastContainerTest.java

```diff
@@ -0,0 +1,106 @@
+package org.testcontainers.examples;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.core.HazelcastInstance;
+import org.junit.AfterClass;
+import org.junit.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.lifecycle.Startables;
+import org.testcontainers.utility.DockerImageName;
+
+import java.util.concurrent.BlockingQueue;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Examples with Hazelcast using both a single container and a cluster with two containers.
+ */
+public class HazelcastContainerTest {
+
+    // Hazelcast values
+    private static final String HZ_IMAGE_NAME = "hazelcast/hazelcast:5.2.0";
+
+    private static final String HZ_CLUSTERNAME_ENV_NAME = "HZ_CLUSTERNAME";
+
+    private static final int DEFAULT_EXPOSED_PORT = 5701;
+
+    private static final String CLUSTER_STARTUP_LOG_MESSAGE_REGEX = ".*Members \\{size:%d.*";
+
+    // Test values
+    private static final String HOST_PORT_SEPARATOR = ":";
+
+    private static final String TEST_QUEUE_NAME = "test-queue";
+
+    private static final String TEST_CLUSTER_NAME = "test-cluster";
+
+    private static final String TEST_VALUE = "Hello!";
+
+    @AfterClass
+    public static void cleanUp() {
+        HazelcastClient.shutdownAll();
+    }
+
+    @Test
+    public void singleHazelcastContainer() throws InterruptedException {
+        try (
+            GenericContainer<?> container = new GenericContainer<>(DockerImageName.parse(HZ_IMAGE_NAME))
+                .withExposedPorts(DEFAULT_EXPOSED_PORT)
+        ) {
+            container.start();
+            assertThat(container.isRunning()).isTrue();
+
+            ClientConfig clientConfig = new ClientConfig();
+            clientConfig
+                .getNetworkConfig()
+                .addAddress(container.getHost() + HOST_PORT_SEPARATOR + container.getFirstMappedPort());
+            HazelcastInstance client = HazelcastClient.newHazelcastClient(clientConfig);
+
+            BlockingQueue<String> queue = client.getQueue(TEST_QUEUE_NAME);
+            queue.put(TEST_VALUE);
+            assertThat(queue.take()).isEqualTo(TEST_VALUE);
+        }
+    }
+
+    @Test
+    public void hazelcastCluster() throws InterruptedException {
+        Network network = Network.newNetwork();
+        try (
+            GenericContainer<?> container1 = new GenericContainer<>(DockerImageName.parse(HZ_IMAGE_NAME))
+                .withExposedPorts(DEFAULT_EXPOSED_PORT)
+                .withEnv(HZ_CLUSTERNAME_ENV_NAME, TEST_CLUSTER_NAME)
+                .waitingFor(Wait.forLogMessage(String.format(CLUSTER_STARTUP_LOG_MESSAGE_REGEX, 1), 1))
```

Comment:
You don't know which one will connect first :D

```suggestion
                .waitingFor(Wait.forLogMessage(String.format(CLUSTER_STARTUP_LOG_MESSAGE_REGEX, 2), 1))
```

Reply:
Oh, sorry, I missed this.

I'm not sure what you mean by I don't know which will connect first - since we're now starting the containers sequentially and waiting for the log to appear,  I think the first container will always connect first, and when the second one connects it will always find the first.

What am I missing? 🧐

Reply:
that was before you made the change. So, no missing anything 😀

Reply:
Cool! Just for reference I changed both in [this commit](https://github.com/testcontainers/testcontainers-java/pull/6117/commits/97d35e77f462d50d044e9824bea65ee6a0ad24c9)